addon/components/new-bourbon-select-field.js

```diff
@@ -0,0 +1,214 @@
+    if (isPresent(this.get('optionEnabledPath'))) {
+      return this.get('optionEnabledPath').replace(/^content\.?/, '');
+    }
+  }),
+
+  internalContent: computed(
+    'content',
+    '_labelPath',
+    '_valuePath',
+    '_enabledPath',
+    function() {
+      return A(
+        this.get('content').map(item => {
+          if (typeof item === 'string' || typeof item === 'number') {
+            return EmberObject.create({
+              label: item.toString(),
+              value: item,
+              enabled: true
+            });
+          } else {
+            let emberItem =
+              typeof item.get == 'function' ? item : EmberObject.create(item);
+
+            let endItem = {};
+            endItem.label = this.get('_labelPath')
+              ? emberItem.get(this.get('_labelPath'))
+              : emberItem.get('label');
+            endItem.value = this.get('_valuePath')
+              ? emberItem.get(this.get('_valuePath'))
+              : emberItem.get('value');
+            endItem.enabled = this.get('_enabledPath')
+              ? emberItem.get(this.get('_enabledPath'))
+              : true;
+
+            // when the value is supposed to be the item
+            if (endItem.value === undefined) endItem.value = emberItem;
+
+            return EmberObject.create(endItem);
+          }
+        })
+      );
+    }
+  ),
+
+  selectedIndex: computed('internalContent', 'value', {
+    get(key) {
+      let valueHolder = this.get('internalContent').findBy(
+        'value',
+        this.get('value')
+      );
+      return this.get('internalContent').indexOf(valueHolder);
+    },
+
+    set(key, value) {
+      this.set(
+        'value',
+        this.get('internalContent')
+          .objectAt(value)
+          .get('value')
+      );
+      return value;
+    }
+  }),
+
+  activeIndex: computed('selectedIndex', {
+    get(key) {
+      return this.get('selectedIndex');
+    },
+
+    set(key, value) {
+      return value;
+    }
+  }),
+
+  selection: computed('selectedIndex', function() {
+    if (this.get('selectedIndex') !== -1) {
+      return this.get('internalContent').objectAt(this.get('selectedIndex'));
+    } else {
+      return null;
+    }
+  }),
+
+  label: computed('selection', 'prompt', function() {
+    if (isPresent(this.get('selection'))) {
+      return this.get('selection.label');
+    } else {
+      return this.get('prompt');
+    }
+  }),
+
+  focusOut() {
+    this.set('showList', false);
+  },
+
+  keyDown(e) {
+    if (e.keyCode === 38) {
+      // up arrow
+      this.moveActiveUp(this.get('activeIndex'));
+    } else if (e.keyCode === 40) {
+      // down arrow
+      this.moveActiveDown(this.get('activeIndex'));
+    } else if (e.keyCode === 13) {
+      //  enter
+      if (this.get('showList')) {
+        this.set('selectedIndex', this.get('activeIndex'));
+        this.set('showList', false);
+        document.activeElement.blur();
```

Comment:
Does this`document.activeElement.blur();` trigger the `focusOut` method above? If so, I don't think we need to set `this.set('showList', false);` in here.

Reply:
good call.  it does trigger the `focusOut`.  going to remove it.  :)